src/testdir/test_increment.vim

```diff
@@ -796,4 +796,54 @@ func Test_increment_special_char()
   call assert_beeps("normal \<C-X>")
 endfunc
 
+" Try incrementing/decrementing a number when nrformats contains unsigned
+func Test_unsigned()
+  set nrformats+=unsigned
+
+  " 18446744073709551615 == pow(2, 64) - 1
+  if 64 == v:numbersize
+    call setline(1, '0')
+    exec "norm! gg0\<C-X>"
+    call assert_equal('18446744073709551615', getline(1))
+
+    call setline(1, '3')
+    exec "norm! gg010\<C-X>"
+    call assert_equal('18446744073709551609', getline(1))
+
+    call setline(1, '-0')
+    exec "norm! gg0\<C-X>"
+    call assert_equal("-18446744073709551615", getline(1))
+
+    call setline(1, '18446744073709551615')
+    exec "norm! gg0\<C-A>"
+    call assert_equal('0', getline(1))
+
+    call setline(1, '18446744073709551613')
+    exec "norm! gg010\<C-A>"
+    call assert_equal('7', getline(1))
+
+    call setline(1, '-18446744073709551615')
+    exec "norm! gg0\<C-A>"
+    call assert_equal('-0', getline(1))
+  endif
+
+  call setline(1, '-1')
+  exec "norm! gg0\<C-X>"
+  call assert_equal('-0', getline(1))
+
+  call setline(1, '-3')
+  exec "norm! gg0\<C-X>"
+  call assert_equal('-2', getline(1))
+
+  call setline(1, '-1')
+  exec "norm! gg0\<C-A>"
+  call assert_equal('-2', getline(1))
+
+  call setline(1, '-3')
+  exec "norm! gg0\<C-A>"
+  call assert_equal('-4', getline(1))
+
+  set nrformats-=unsigned
```

Comment:
This restoring 'nrformats' is no need because it is set to default in SetUp().

Reply:
I fixed it. Thank you for reviewing this PR.

Reply:
FWIW: I prefer explicitly resetting of the `nrformats` just to make sure it is explicit and when a somebody copy pastes this as a new test, it does not depend on the Setup and TearDown function.